src/util/widgethelper.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include <QScreen>
+
+namespace mixxx {
+namespace WidgetHelper {
+/// Returns an adjusted upper left point for displaying the popup
+/// with the given size on the screen, shifting the popup if it would go off
+/// the right or bottom edges of the screen.
+QPoint mapPopupToScreen(
```

Comment:
Needs to be declared as `inline` or moved into a .cpp file. Otherwise the linker will complain about duplicate symbols when reused.